File: frontend/src/types/prediction.ts
export interface ProductVelocity {
  product_id: number
  product_name: string
  sku: string
  category: string
  current_stock: number
  units_sold: number
  daily_velocity: number
  monthly_forecast: number
  classification: string
}

export interface ReorderRecommendation {
  product_id: number
  product_name: string
  sku: string
  current_stock: number
  min_stock: number
  max_stock: number
  avg_daily_sales: number
  days_of_stock_left: number | null
  recommended_qty: number
  priority: string
  reason: string
}

export interface NextMonthForecast {
  product_id: number
  product_name: string
  sku: string
  current_stock: number
  predicted_demand: number
  projected_stock: number
  shortfall: number
  status: string
}

export interface PredictionDashboard {
  analysis_period_days: number
  fast_moving_count: number
  slow_moving_count: number
  reorder_needed_count: number
  total_predicted_monthly_demand: number
  fast_movers: ProductVelocity[]
  slow_movers: ProductVelocity[]
  reorder_recommendations: ReorderRecommendation[]
  next_month_forecast: NextMonthForecast[]
}

File: backend/app/schemas/prediction.py
from pydantic import BaseModel, Field


class ProductVelocity(BaseModel):
    product_id: int
    product_name: str
    sku: str
    category: str
    current_stock: int
    units_sold: int
    daily_velocity: float
    monthly_forecast: int
    classification: str  # fast_moving | slow_moving | normal


class ReorderRecommendation(BaseModel):
    product_id: int
    product_name: str
    sku: str
    current_stock: int
    min_stock: int
    max_stock: int
    avg_daily_sales: float
    days_of_stock_left: float | None
    recommended_qty: int
    priority: str  # critical | high | medium | low
    reason: str


class NextMonthForecast(BaseModel):
    product_id: int
    product_name: str
    sku: str
    current_stock: int
    predicted_demand: int
    projected_stock: int
    shortfall: int
    status: str  # sufficient | reorder_needed | overstock_risk


class PredictionDashboard(BaseModel):
    analysis_period_days: int
    fast_moving_count: int
    slow_moving_count: int
    reorder_needed_count: int
    total_predicted_monthly_demand: int
    fast_movers: list[ProductVelocity]
    slow_movers: list[ProductVelocity]
    reorder_recommendations: list[ReorderRecommendation]
    next_month_forecast: list[NextMonthForecast]
